File: app/src/main/java/vn/loitp/app/activity/customviews/button/autosizebutton/AutoSizeButtonActivity.java
package vn.loitp.app.activity.customviews.button.autosizebutton;

import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;

import com.core.base.BaseFontActivity;
import com.core.utilities.LActivityUtil;
import com.core.utilities.LScreenUtil;
import com.views.LToast;
import com.views.autosize.imagebuttonwithsize.ImageButtonWithSize;

import vn.loitp.app.R;

public class AutoSizeButtonActivity extends BaseFontActivity implements OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        findViewById(R.id.bt_rotate).setOnClickListener(this);

        ImageButtonWithSize bt0 = findViewById(R.id.bt_0);
        bt0.setPortraitSizeWInDp(50);
        bt0.setPortraitSizeHInDp(50);
        bt0.setLandscapeSizeWInDp(250);
        bt0.setLandscapeSizeHInDp(250);
        bt0.setOnClickListener(this);

        ImageButtonWithSize bt1 = findViewById(R.id.bt_1);
        bt1.setPortraitSizeWInDp(150);
        bt1.setPortraitSizeHInDp(150);
        bt1.setLandscapeSizeWInDp(100);
        bt1.setLandscapeSizeHInDp(100);
        bt1.setOnClickListener(this);

        ImageButtonWithSize bt2 = findViewById(R.id.bt_2);
        bt2.setPortraitSizeWInPx(LScreenUtil.INSTANCE.getScreenWidth());
        bt2.setPortraitSizeHInPx(LScreenUtil.INSTANCE.getScreenWidth() / 10);
        bt2.setLandscapeSizeWInPx(LScreenUtil.INSTANCE.getScreenWidth() / 2);
        bt2.setLandscapeSizeHInPx(LScreenUtil.INSTANCE.getScreenWidth() / 2);
        bt2.setOnClickListener(this);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_autosize_button;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.bt_rotate:
                LActivityUtil.toggleScreenOritation(getActivity());
                break;
            case R.id.bt_0:
            case R.id.bt_1:
            case R.id.bt_2:
                LToast.show(getActivity(), "Click");
                break;
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/button/autosizebutton/AutoSizeButtonActivity.kt
package vn.loitp.app.activity.customviews.button.autosizebutton

import android.os.Bundle
import android.view.View
import com.annotation.IsFullScreen
import com.annotation.LayoutId
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.core.utilities.LActivityUtil
import com.core.utilities.LScreenUtil
import kotlinx.android.synthetic.main.activity_button_autosize.*
import vn.loitp.app.R

@LayoutId( R.layout.activity_button_autosize)
@LogTag("AutoSizeButtonActivity")
@IsFullScreen(false)
class AutoSizeButtonActivity : BaseFontActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        btRotate.setOnClickListener(this)

        bt0.setPortraitSizeWInDp(50f)
        bt0.setPortraitSizeHInDp(50f)
        bt0.setLandscapeSizeWInDp(250f)
        bt0.setLandscapeSizeHInDp(250f)
        bt0.setOnClickListener(this)

        bt1.setPortraitSizeWInDp(150f)
        bt1.setPortraitSizeHInDp(150f)
        bt1.setLandscapeSizeWInDp(100f)
        bt1.setLandscapeSizeHInDp(100f)
        bt1.setOnClickListener(this)

        bt2.setPortraitSizeWInPx(LScreenUtil.screenWidth)
        bt2.setPortraitSizeHInPx(LScreenUtil.screenWidth / 10)
        bt2.setLandscapeSizeWInPx(LScreenUtil.screenWidth / 2)
        bt2.setLandscapeSizeHInPx(LScreenUtil.screenWidth / 2)
        bt2.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v) {
            btRotate -> LActivityUtil.toggleScreenOritation(this)
            bt0, bt1, bt2 -> showShort("Click")
        }
    }
}
